core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1313,17 +1312,14 @@ public Builder setRef(String name, SnapshotRef ref) {
       Snapshot snapshot = snapshotsById.get(snapshotId);
       ValidationException.check(
           snapshot != null, "Cannot set %s to unknown snapshot: %s", name, snapshotId);
-      if (isAddedSnapshot(snapshotId)) {
-        this.lastUpdatedMillis = snapshot.timestampMillis();
-      }
 
       if (SnapshotRef.MAIN_BRANCH.equals(name)) {
         this.currentSnapshotId = ref.snapshotId();
         if (lastUpdatedMillis == null) {
           this.lastUpdatedMillis = System.currentTimeMillis();
         }
 
-        snapshotLog.add(new SnapshotLogEntry(lastUpdatedMillis, ref.snapshotId()));
+        snapshotLog.add(new SnapshotLogEntry(snapshot.timestampMillis(), snapshot.snapshotId()));
```

Comment:
I don't think that this is correct. Whether to use the snapshot's timestamp in the log entry should still depend on whether the operation is a rollback or an added snapshot, so this needs to continue to check `isAddedSnapshot`.

I don't think that this PR should change any logic other than the snapshot used for table metadata.

Reply:
Thanks ryan for catching this, rollback to an existing snapshot will not reuse its creation timestamp in snapshot log entry totally make senses, as the timestamp in HistoryEntry reflects the time of the change https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/api/src/main/java/org/apache/iceberg/HistoryEntry.java#L30-L31